controllers/namespace/namespace_controller.go

```diff
@@ -1,108 +1,89 @@
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	managedv1alpha1 "github.com/openshift/rbac-permissions-operator/api/v1alpha1"
 )
 
 var log = logf.Log.WithName("controller_namespace")
 
-/**
-* USER ACTION REQUIRED: This is a scaffold file intended for the user to modify with their own Controller
-* business logic.  Delete these comments after modifying this file.*
- */
-
-// Add creates a new Namespace Controller and adds it to the Manager. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
-func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
-}
-
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileNamespace{Client: mgr.GetClient(), Scheme: mgr.GetScheme()}
-}
-
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New("namespace-controller", mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
-	}
-
-	// Watch for changes to primary resource Namespace
-	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-// blank assignment to verify that ReconcileNamespace implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcileNamespace{}
-
-// ReconcileNamespace reconciles a Namespace object
-type ReconcileNamespace struct {
-	// This client, initialized using mgr.Client() above, is a split client
-	// that reads objects from the cache and writes to the apiserver
-	Client client.Client
+// NamespaceReconciler reconciles a Namespace object
+type NamespaceReconciler struct {
+	client.Client
 	Scheme *runtime.Scheme
 }
 
-// Reconcile reads that state of the cluster for a Namespace object and makes changes based on the state read
-// and what is in the Namespace.Spec
-// TODO(user): Modify this Reconcile function to implement your Controller logic.  This example creates
-// a Pod as an example
-// Note:
-// The Controller will requeue the Request to be processed again if the returned error is non-nil or
-// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileNamespace) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+//+kubebuilder:rbac:groups=managed.openshift.io,resources=subjectpermissions,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=managed.openshift.io,resources=subjectpermissions/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=managed.openshift.io,resources=subjectpermissions/finalizers,verbs=update
+//+kubebuilder:rbac:groups=core,resources=namespaces,verbs=get;list
```

Comment:
Am curious about the rbac markers here. We actually using it? IIUC, we have to prepare the manifests separately after the markers are in place but during the migration we also prefer maintaining the previous rbac definitions as it is. Can you confirm?

Reply:
Yes, I think the rbac manifests are not coming from those markers, what we have currently is a clusterrole with much more wider permission.
But those markers are generated via the tools(some of them), do you think we should just remove those lines?

Reply:
Sorry, not aware if some tools generating it or not. For MUO didn't come across those markers being created automatically at all. During OAO work, we noticed that separate `make` commands would need to be run to create respective rbac yaml files from these markers but we already have them as part of repo. So I thought the operator would work the same for us without these markers too. It would be tested and verified though.